--- a/drivers/net/wireless/rt2x00/rt2800lib.c	2014-10-01 22:09:41.000000000 +0200
+++ b/drivers/net/wireless/rt2x00/rt2800lib.c	2014-10-06 18:27:05.000000000 +0200
@@ -6832,6 +6832,7 @@
 	case RT3883:
 		rt2800_init_bbp_3883(rt2x00dev);
 		return;
+	case RT5362:
 	case RT5390:
 	case RT5392:
 		if (rt2x00dev->chip.rf == RF7620)
@@ -8380,6 +8381,7 @@
 		else
 			rt2800_init_rfcsr_5390(rt2x00dev);
 		break;
+	case RT5362:
 	case RT5392:
 		rt2800_init_rfcsr_5392(rt2x00dev);
 		break;
